wlauto/workloads/googlephotos/__init__.py

```diff
@@ -0,0 +1,104 @@
+
+    name = 'googlephotos'
+    package = 'com.google.android.apps.photos'
+    version = '1.21.0.123444480'  # apk version
+    activity = 'com.google.android.apps.photos.home.HomeActivity'
+    view = [package + '/com.google.android.apps.consumerphotoeditor.fragments.ConsumerPhotoEditorActivity',
+            package + '/com.google.android.apps.photos.home.HomeActivity',
+            package + '/com.google.android.apps.photos.localmedia.ui.LocalPhotosActivity',
+            package + '/com.google.android.apps.photos.onboarding.AccountPickerActivity',
+            package + '/com.google.android.apps.photos.onboarding.IntroActivity']
+    description = """
+    A workload to perform standard productivity tasks with Google Photos. The workload carries out
+    various tasks, such as browsing images, performing zooms, and post-processing the image.
+
+    Test description:
+    1. Four images are copied to the device
+    2. The application is started in offline access mode
+    3. Gestures are performed to swipe between images and pinch zoom in and out of the selected
+       image
+    4. The colour of a selected image is edited by selecting the colour menu, incrementing the
+       colour, resetting the colour and decrementing the colour using the seek bar.
+    5. A crop test is performed on a selected image.  UiAutomator does not allow the selection of
+       the crop markers so the image is tilted positively, reset and then negatively to get a
+       similar cropping effect.
+    6. A rotate test is performed on a selected image, rotating anticlockwise 90 degrees, 180
+       degrees and 270 degrees.
+    """
+
+    default_test_images = [
+        'uxperf_1200x1600.png', 'uxperf_1600x1200.jpg',
+        'uxperf_2448x3264.png', 'uxperf_3264x2448.jpg',
+    ]
+
+    parameters = [
+        Parameter('markers_enabled', kind=bool, default=True,
+                  description="""
+                  If ``True``, UX_PERF action markers will be emitted to logcat during
+                  the test run.
+                  """),
+        Parameter('test_images', kind=list_of_strings, default=default_test_images,
+                  description="""
+                  A list of four image files to be pushed to the device.
+                  Absolute file paths may be used but tilde expansion must be escaped.
+                  """),
+    ]
+
+    def validate(self):
+        super(Googlephotos, self).validate()
+        self.uiauto_params['package'] = self.package
+        self.uiauto_params['output_dir'] = self.device.working_directory
+        self.uiauto_params['markers_enabled'] = self.markers_enabled
+
+        if len(self.test_images) != 4 or len(self.test_images) != len(set(self.test_images)):
+            message = "This workload requires four unique test images"
+            raise ValidationError(message)
+
+        for image in self.test_images:
+            if not image.endswith(('jpg', 'jpeg', 'png')):
```

Comment:
Better done as:

     if os.path.splitext(image)[1] not in ['.jpg', '.jpeg', '.png']: